lib/scnnr/polling_utils.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Scnnr
+  class PollingUtils
+    MAX_TIMEOUT_INTERVAL = 25
+
+    def self.start(timeout_at, &block)
+      block_value = block.call
+      return block_value if block_value.finished?
+
+      raise TimeoutError.new('Polling timed out paitently', block_value) unless Time.now.utc < timeout_at
```

Comment:
If we change the logic around `timeout` parameter in this `timeout_at` way, this PR is not "Refactor" anymore because it changes the behavior. Because there should be time to be taken to preparing a request, establishing a connection, sending the request, etc.
If this is not intentional, could we revert this?

Reply:
This is a `proposal` since I thought this is coding test, which I would like to discuss with you later.